lib/ansible/modules/identity/keycloak/keycloak_clienttemplate.py

```diff
@@ -41,8 +41,9 @@
             - State of the client template
             - On C(present), the client template will be created (or updated if it exists already).
             - On C(absent), the client template will be removed if it exists
+            - On C(dump), the client template will be read and provided in the "existing" return value only.
         required: false
-        choices: ['present', 'absent']
+        choices: ['present', 'absent', 'dump']
```

Comment:
We historically have not permitted a `state` used for retrieval, as it is technically not a state.  Our recommendation for any retrieval, is to implement that code in a lookup plugin, as opposed to a module.

Reply:
I will remove that explicit state again; implicitly you get the same behavior when using check mode, i.e. the existing settings are provided in "existing," basically as a side-effect.

If I understand things correctly (and I may well not), a lookup plugin is executed on the node executing the playbook, not on the node specified in the playbook. The only way to obtain state remotely would appear to be a regular module.